.github/workflows/handler-pr-close.yml

```
@@ -83,27 +88,88 @@ jobs:
+        if: steps.get-workspaces-google-private-active-active.outputs.WORSPACE_PRI_AA_EXISTS == '1'
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        run: terraform destroy -auto-approve -input=false -no-color
+
+    # this step will delete workspace for this PR
+      - name: Terraform Delete Workspace
+        id: destroy-workspace
+        if: steps.get-workspaces-google-private-active-active.outputs.WORSPACE_PRI_AA_EXISTS == '1' && steps.destroy.outcome == 'success'
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        env:
+          TFE_ORGANIZATION: ${{ secrets.$TFE_ORGANIZATION }}
+          TFE_TOKEN: ${{ secrets.TFE_TOKEN }}
+        run: >-
+          curl \ --header "Authorization: Bearer $TFE_TOKEN" \ --header "Content-Type: application/vnd.api+json" \ --request DELETE \  https://app.terraform.io/api/v2/organizations/$TFE_ORGANIZATION/workspaces/google-private-active-active-${{ github.event.pull_request.number }}
+    
+    # this step will check if workspace is created for google-public-active-active for this PR
+      - name: Check if workspace for google-public-active-active exists
+        id: check-workspaces-google-public-active-active
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        run: >-
+          echo "::set-output name=WORSPACE_PUB_AA_EXISTS::$(curl -s \  --header "Authorization: Bearer $TFE_TOKEN" \  --header "Content-Type: application/vnd.api+json" \  https://app.terraform.io/api/v2/organizations/$TFE_ORGANIZATION/workspaces | jq -r '.data[] | .attributes | .name' | grep -cx google-public-active-active-${{ github.event.pull_request.number }})"
+        env:
+          TFE_ORGANIZATION: ${{ secrets.TFE_ORGANIZATION }}
+          TFE_TOKEN: ${{ secrets.TFE_TOKEN }}
+
+    # this step will delete workspace-resources if exists for this PR
+      - name: Terraform Destroy
+        id: destroy
+        if: steps.get-workspaces-google-public-active-active.outputs.WORSPACE_PUB_AA_EXISTS == '1'
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        run: terraform destroy -auto-approve -input=false -no-color
+
+    # this step will delete workspace for this PR
+      - name: Terraform Delete Workspace
+        id: destroy-workspace
+        if: steps.get-workspaces-google-public-active-active.outputs.WORSPACE_PUB_AA_EXISTS == '1' && steps.destroy.outcome == 'success'
+        working-directory: ${{ env.WORK_DIR_PATH }}
+        env:
+          TFE_ORGANIZATION: ${{ secrets.$TFE_ORGANIZATION }}
           TFE_TOKEN: ${{ secrets.TFE_TOKEN }}
-        run: terraform workspace delete "${{ steps.get-workspaces-google-private-tcp-active-active.outputs.WORSPACE_TCP_AA }}" -no-color
-   
+        run: >-
+          curl \ --header "Authorization: Bearer $TFE_TOKEN" \ --header "Content-Type: application/vnd.api+json" \ --request DELETE \  https://app.terraform.io/api/v2/organizations/$TFE_ORGANIZATION/workspaces/google-public-active-active-${{ github.event.pull_request.number }}
```

I don't believe you need the backslash characters here since this is on a single line.

```suggestion
          curl --header "Authorization: Bearer $TFE_TOKEN" --header "Content-Type: application/vnd.api+json" --request DELETE https://app.terraform.io/api/v2/organizations/$TFE_ORGANIZATION/workspaces/google-public-active-active-${{ github.event.pull_request.number }}
```

---

updated